WORKSPACE

```
@@ -193,6 +193,7 @@ rules_proto_toolchains()
 load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
 load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies", "go_repository")
 load("//:deps.bzl", "go_dependencies")
+load("//:linters.bzl", "linter_dependencies")
```

```suggestion
load("//:linters_deps.bzl", "linter_dependencies")
```

A bit nitpicky, but I think that will help if you don't mind renaming this.

---

I agree completely actually.

TL;DR: Bazel actually recommends that file targets be colocated where they are used, meaning, since `nogo_config.json` is used in the root BUILD.bazel in the `nogo` rule, it's better to have the file in the root and to refer to it as `//:nogo_config.json`

 I did go down a mini rabbit hole in trying to have all of this co-located in the //dev/linters package but I had trouble referencing the `nogo_config.json` file in the repo root BUILD.bazel. 

`bazel query` showed the target existed:
```
bazel query '//dev/linters:*'
//dev/linters:BUILD.bazel
//dev/linters:nogo_config.json
```
`bazel cquery //dev/linters:nogo_config.json --output=files` showed the file:
```
bazel cquery //dev/linters:nogo_config.json --output=files
INFO: Analyzed target //dev/linters:nogo_config.json (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
dev/linters/nogo_config.json
INFO: Elapsed time: 0.236s, Critical Path: 0.00s
INFO: 0 processes.
INFO: Build completed successfully, 0 total actions
```

but once you refer to the file in the config attribute of the nogo rule, you would get:
```
bazel build //cmd/frontend
ERROR: /Users/william/code/sourcegraph/BUILD.bazel:205:5: no such target '//dev/linters:nogo_config.json)': target 'nogo_config.json)' not declared in package 'dev/linters' defined by /Users/william/code/sourcegraph/dev/linters/BUILD.bazel (did you mean 'nogo_config.json'? Tip: use `query "//dev/linters:*"` to see all the targets in that package) and referenced by '//:sg_nogo_actual'
```

I also tried `file_group` and referring to the location with `$(location )` but no 🎲